lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -1303,6 +1304,88 @@ class TTIRGridSet : public impl::TTIRGridSetBase<TTIRGridSet> {
       func.setType(newFuncType);
     });
   }
+
+  void processReshardedEdges(const std::unordered_set<Edge> &reshardedEdges) {
+    // Insert reshard ops here based on results of sharding analysis.
+    //
+    for (const Edge &edge : reshardedEdges) {
+      Operation *producerOp = edge.producerOp;
+      Operation *consumerOp = edge.consumerOp;
+
+      // If producerOp is a ToLayoutOp, adjust its output layout(update inplace)
+      // to reflect consumerOp's output layout. If producerOp is not a
+      // ToLayoutOp, insert a ToLayoutOp in between producerOp and consumerOp.
+      //
+      if (llvm::isa<ttir::ToLayoutOp>(producerOp)) {
+        ttir::ToLayoutOp toLayoutOp = llvm::cast<ttir::ToLayoutOp>(producerOp);
+        LayoutAttr consumerOpOutputLayout = mlir::cast<LayoutAttr>(
+            mlir::cast<RankedTensorType>(consumerOp->getResult(0).getType())
+                .getEncoding());
+
+        RankedTensorType toLayoutOpTensorType =
+            mlir::cast<RankedTensorType>(toLayoutOp.getResult().getType());
+        llvm::ArrayRef<int64_t> toLayoutOpTensorShape =
+            toLayoutOpTensorType.getShape();
+        LayoutAttr toLayoutOpLayout =
+            mlir::cast<LayoutAttr>(toLayoutOpTensorType.getEncoding());
+
```

Comment:
@mbezuljTT We need API call - for op type, input&output tensor shapes, output sharding layout which input sharding layout is legal.

Reply:
This comes down to, given sharding type on output, which sharding type is valid on input_0.

Reply:
This type of bs 
![image](https://github.com/user-attachments/assets/0f36a00f-3849-4f7a-8e59-31bbd9053587)